addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -253,62 +255,156 @@ def test_post_wo_access(self):
+                            "type": "activity",
+                            "view_type": "list",
+                            "overdue_count": 0,
+                            "planned_count": 0,
+                            "today_count": len(expected_other_activities),
+                            "total_count": len(expected_other_activities),
+                            "id": self.env["ir.model"]._get_id(other_activities_model_name),
+                            "model": other_activities_model_name,
+                            "name": "Other activities",
+                            "icon": "/mail/static/description/icon.png",
+                            "activity_ids": set(expected_other_activities.ids),
+                        },
+                        {
+                            **actual_activity_group,
+                            # To compare regardless the order
+                            "activity_ids": set(actual_activity_group['activity_ids']),
+                        }
+                    )
+                test_model_name = 'mail.test.multi.company.with.activity'
+                self.assertIn(test_model_name, actual_activity_groups_by_model)
+                self.assertDictEqual(
+                    {
+                        "type": "activity",
+                        "view_type": "list",
+                        "overdue_count": 0,
+                        "planned_count": 0,
+                        "today_count": len(expected_test_model_activities),
+                        "total_count": len(expected_test_model_activities),
+                        "id": self.env["ir.model"]._get_id(test_model_name),
+                        "model": test_model_name,
+                        "name": "Test Multi Company Mail With Activity",
+                        "icon": "/base/static/description/icon.png",
+                    },
+                    actual_activity_groups_by_model[test_model_name])
+                allowed_user_company_records = _mock_filter_access_rules(
+                    (self.env[test_model_name]
+                    ).with_context(allowed_company_ids=allowed_company_ids).with_user(user).search([
+                        ('id', 'in', test_records_all.ids),
+                        ('activity_user_id', '=', user.id),
+                    ]),
+                    'read')
+                self.assertEqual(
+                    (self.env['mail.activity']
+                    ).with_user(user).with_context(allowed_company_ids=allowed_company_ids).search([
+                        ('res_model', '=', test_model_name),
+                        ('res_id', 'in', allowed_user_company_records.ids),
+                        ('user_id', '=', user.id),
+                    ]),
+                    expected_test_model_activities)
+                self.assertEqual(len(allowed_user_company_records), len(expected_test_model_activities),
+                                 'One scheduled activity per record per user')
```

Comment:
I don't really understand what this part is testing.

Looks like we're entirely testing our mock + the regular search behavior based on companies, which seems to have little use in the test context? And outside of its responsibilities? 

Must me missing something.